c_src/enacl_nif.c

```diff
@@ -485,6 +495,65 @@ ERL_NIF_TERM enif_crypto_sign_open(ErlNifEnv *env, int argc, ERL_NIF_TERM const
 	}
 }
 
+/*
+int crypto_sign_detached(unsigned char *sig, unsigned long long *siglen,
+                         const unsigned char *m, unsigned long long mlen,
+                         const unsigned char *sk);
+ */
+static
+ERL_NIF_TERM enif_crypto_sign_detached(ErlNifEnv* env, int argc, ERL_NIF_TERM const argv[]) {
+        ErlNifBinary m, sk, sig;
+        unsigned long long siglen;
+
+        if (
+            (argc != 2) ||
+            (!enif_inspect_iolist_as_binary(env, argv[0], &m)) ||
+            (!enif_inspect_binary(env, argv[1], &sk))) {
+            return enif_make_badarg(env);
+        }
+
+        if (sk.size != crypto_sign_SECRETKEYBYTES) {
+            return enif_make_badarg(env);
+        }
+
+        if (!enif_alloc_binary(crypto_sign_BYTES, &sig)) {
+            return nacl_error_tuple(env, "alloc_failed");
+        }
+
+        crypto_sign_detached(sig.data, &siglen, m.data, m.size, sk.data);
+
+        return enif_make_sub_binary(env, enif_make_binary(env, &sig), 0, siglen);
```

Comment:
This should probably just be `enif_make_binary(env, &sig)`, as we don't need to create a sub_binary for it.